source/spirv_definition.h

```diff
@@ -39,6 +43,16 @@ typedef uint64_t spv_capability_mask_t;
 #define SPV_CAPABILITY_AS_MASK(capability) \
   (spv_capability_mask_t(1) << (capability))
 
+namespace libspirv {
+template <typename Functor>
+inline void ForEach(spv_capability_mask_t capabilities, Functor f) {
+  for (int cap = SpvCapabilityMatrix;
+       cap <= SpvCapabilityStorageImageWriteWithoutFormat; ++cap)
```

Comment:
It would be nice if there were a "kCapabilityMaxValue" we could reuse.
There's a kCapabilitiesMaxValue static size_t variable in validate_types.cpp

Maybe it's time to put that definition in this header.  
I can wait for another CL, however.

Also, I'd much rather have that max value be defined by some file in the SPIR-V registry, either spirv.h or something generated from spirv.json.  So another reason this can wait if you want.